src/http/cookie.cr

```diff
@@ -99,16 +124,14 @@ module HTTP
         match = header.match(SetCookieString)
         return unless match
 
-        expires = if max_age = match["max_age"]?
-                    Time.now + max_age.to_i.seconds
-                  else
-                    parse_time(match["expires"]?)
-                  end
+        expires = parse_time(match["expires"]?)
+        max_age = match["max_age"]? ? match["max_age"].to_i.seconds : nil
```

Comment:
```cr
max_age = match["max_age"]?.try(&.to_i.seconds)
```

Reply:
This suggestion seems better than the current code.